com.unity.multiplayer.mlapi/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -219,121 +383,248 @@ internal void OnSceneSwitch(uint sceneIndex, Guid switchSceneGuid, Stream object
+                        OnSceneEvent?.Invoke(new SceneEvent()
+                        {
+                            AsyncOperation = SceneManager.UnloadSceneAsync(loadedSceneName),
+                            SceneEventType = SceneEventData.SceneEventTypes.S2C_Unload,
+                            LoadSceneMode = LoadSceneMode.Additive,
+                            SceneName = loadedSceneName,
+                            ClientId = m_NetworkManager.LocalClientId
+                        });
                     }
                 }
+                m_ScenesLoaded.Clear();
+
+                // Move ALL NetworkObjects to the temp scene
+                MoveObjectsToDontDestroyOnLoad();
+            }
+
+            // The Condition: While a scene is asynchronously loaded in single loading scene mode, if any new NetworkObjects are spawned
+            // they need to be moved into the do not destroy temporary scene
+            // When it is set: Just before starting the asynchronous loading call
+            // When it is unset: After the scene has loaded, the PopulateScenePlacedObjects is called, and all NetworkObjects in the do
+            // not destroy temporary scene are moved into the active scene
+            if (SceneEventData.LoadSceneMode == LoadSceneMode.Single)
+            {
+                IsSpawnedObjectsPendingInDontDestroyOnLoad = true;
             }
+
+            var sceneLoad = SceneManager.LoadSceneAsync(sceneName, SceneEventData.LoadSceneMode);
+            sceneLoad.completed += asyncOp2 => OnSceneLoaded(sceneName);
+
+            OnSceneEvent?.Invoke(new SceneEvent()
+            {
+                AsyncOperation = sceneLoad,
+                SceneEventType = SceneEventData.SceneEventType,
+                LoadSceneMode = SceneEventData.LoadSceneMode,
+                SceneName = sceneName,
+                ClientId = m_NetworkManager.LocalClientId
+            });
         }
 
-        private void OnSceneLoaded(Guid switchSceneGuid, Stream objectStream)
+        /// <summary>
+        /// Client and Server:
+        /// Generic on scene loaded callback method to be called upon a scene loading
+        /// </summary>
+        private void OnSceneLoaded(string sceneName)
         {
-            CurrentActiveSceneIndex = SceneNameToIndex[s_NextSceneName];
-            var nextScene = SceneManager.GetSceneByName(s_NextSceneName);
-            SceneManager.SetActiveScene(nextScene);
+            var nextScene = SceneManager.GetSceneByName(sceneName);
+            if (SceneEventData.LoadSceneMode == LoadSceneMode.Single)
+            {
+                SceneManager.SetActiveScene(nextScene);
+            }
+
+            if (!m_ScenesLoaded.Contains(sceneName))
+            {
+                m_ScenesLoaded.Add(sceneName);
+            }
+            else
+            {
+                throw new Exception($"{sceneName} is being loaded twice?!");
```

Comment:
Do we want a more specific error message here (as unlikely it is to happen)

Reply:
Yeah, I think the wording could be improved.  Will see if I can spruce that up a bit.

Reply:
Ok, I just removed that exception and one other that was being thrown under the same condition.  It shouldn't throw an exception on that (was using it for debugging), and that will be removed in the next PR where Netcode scene registration is based off of scenes in the build list and not NetworkManager registration (the whole name tracking thing was to keep track of the delta in "registered scene names" loaded).